server/monitor/src/main/java/org/apache/accumulo/monitor/rest/resources/TracesResource.java

```diff
@@ -0,0 +1,356 @@
+  }
+
+  @Path("/listType/{type}/{minutes}")
+  @GET
+  public TraceType getTracesType(@PathParam("type") String type, @PathParam("minutes") int minutes) throws Exception {
+
+    TraceType typeTraces = new TraceType(type);
+
+    Entry<Scanner,UserGroupInformation> pair = getScanner();
+    final Scanner scanner = pair.getKey();
+    if (scanner == null) {
+      return null;
+    }
+
+    Range range = getRangeForTrace(minutes);
+
+    scanner.setRange(range);
+
+    if (null != pair.getValue()) {
+      pair.getValue().doAs(new PrivilegedAction<Void>() {
+        @Override
+        public Void run() {
+          for (Entry<Key,Value> entry : scanner) {
+            RemoteSpan span = TraceFormatter.getRemoteSpan(entry);
+
+            if (span.description.equals(type)) {
+              typeTraces.addTrace(new TracesForTypeInformation(span));
+            }
+          }
+          return null;
+        }
+      });
+    } else {
+      for (Entry<Key,Value> entry : scanner) {
+        RemoteSpan span = TraceFormatter.getRemoteSpan(entry);
+        if (span.description.equals(type)) {
+          typeTraces.addTrace(new TracesForTypeInformation(span));
+        }
+
+      }
+    }
+
+    return typeTraces;
+  }
+
+  @Path("/show/{id}")
+  @GET
+  public TraceList getTracesType(@PathParam("id") String id) throws Exception {
+    TraceList traces = new TraceList(id);
+
+    if (id == null) {
+      return null;
+    }
+
+    Entry<Scanner,UserGroupInformation> entry = getScanner();
+    final Scanner scanner = entry.getKey();
+    if (scanner == null) {
+      return null;
+    }
+
+    Range range = new Range(new Text(id));
+    scanner.setRange(range);
+    final SpanTree tree = new SpanTree();
+    long start;
+
+    if (null != entry.getValue()) {
+      start = entry.getValue().doAs(new PrivilegedAction<Long>() {
+        @Override
+        public Long run() {
+          return addSpans(scanner, tree, Long.MAX_VALUE);
+        }
+      });
+    } else {
+      start = addSpans(scanner, tree, Long.MAX_VALUE);
+    }
+
+    final long finalStart = start;
+    Set<Long> visited = tree.visit(new SpanTreeVisitor() {
+      @Override
+      public void visit(int level, RemoteSpan parent, RemoteSpan node, Collection<RemoteSpan> children) {
+        traces.addTrace(addTraceInformation(level, node, finalStart));
+      }
+    });
+    tree.nodes.keySet().removeAll(visited);
+    if (!tree.nodes.isEmpty()) {
+      for (RemoteSpan span : TraceDump.sortByStart(tree.nodes.values())) {
+        traces.addTrace(addTraceInformation(0, span, finalStart));
+      }
+    }
+
+    return traces;
+  }
+
+  private static TraceInformation addTraceInformation(int level, RemoteSpan node, long finalStart) {
+
+    boolean hasData = node.data != null && !node.data.isEmpty();
+    boolean hasAnnotations = node.annotations != null && !node.annotations.isEmpty();
+
+    AddlInformation addlData = new AddlInformation();
+
+    if (hasData || hasAnnotations) {
```

Comment:
I see it now, I took it from the current monitor and it is needed there, but not here, thanks!